src/main/scala/carldata/series/TimeSeries.scala

```diff
@@ -128,6 +127,22 @@ object TimeSeries {
     TimeSeries(index.take(vs.length), vs)
   }
 
+  def join[V](ts: Seq[TimeSeries[V]]): TimeSeries[Seq[V]] = {
+    def joinR(xs: TimeSeries[Seq[V]], ys: Seq[TimeSeries[V]]): Seq[(LocalDateTime, (Seq[V]))] = {
```

Comment:
please add @tailrec annotation if you need recursion.

But:

In this case I think that it is better to use foldLeft or reduceLeft operation.